src/autoschedulers/adams2019/AutoSchedule.cpp

```diff
@@ -305,45 +266,47 @@ IntrusivePtr<State> optimal_schedule_pass(FunctionDAG &dag,
             q.emplace(std::move(s));
         };
 
+#ifdef HALIDE_AUTOSCHEDULER_ALLOW_CYOS
```

Comment:
Why a separate ifdef for this one? CYOS is for debugging why a schedule that you want doesn't seem to be in the search space, or is being rejected based on cost. It'd be pretty annoying to have to recompile the autoscheduler to use it.

Reply:
I argue that literally no one other than a developer/maintainer of this autoscheduler will ever use this feature; having an interactive mode in what is otherwise a command-driven tool seems like an odd fit at best, and a security risk at worse. But maybe I'm wrong about usage? I'd love to hear your experience regarding who has used this feature (and how and why) in the past.

Reply:
Actually CYOS should possibly just be moved into the params struct

Reply:
(Race condition on our previous comments)

Hrm, you're probably right.

Reply:
Does that mean it's ok as-is, or do you want revisions?

Reply:
Maybe just put it behind the legacy api flag instead of its own separate thing

Reply:
Hmm, well my plan was to remove the legacy API flag entirely sometime in Halide 16, but I suspect you'd like to keep this for future experimentation?

Reply:
Fair enough, I guess it's fine as-is